### PR TITLE
FIx version generation for Release and Nightly builds

### PR DESCRIPTION
**Pull request by alexbakharew**
### Scope & Purpose

CMake will honor BUILD_REPO_INFO variable from Oskar and set proper version string for CE and EE

Change version generation for build-repository and enterprise-build-repository in CMake in the following way:

- For Release builds:
    refs/tags/vX.Y.Z (<COMMIT_SHORT>)

- For Nightly builds:
    refs/head/X.Y (<COMMIT_SHORT>)

- For custom branches (e.g. when we build neither Release nor Nightly packages)
     refs/<BRANCH_NAME> (<COMMIT_SHORT>)

Related Oskar PR: https://github.com/arangodb/oskar/pull/528

- [ ] :hankey: Bugfix
- [X] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [X] Backport for 3.11: *(Please link PR)*
  - [X] Backport for 3.10: *(Please link PR)*
  - [X] Backport for 3.9: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [X] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/QA-446
- [ ] Design document:

### CODE REVIEW

**Comment by alexbakharew** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/31003/

**Review by dothebart — APPROVED** [review]

**Review by Dronplane — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -291,44 +291,102 @@ configure_file(
...
+  elseif(BUILD_REPO_INFO STREQUAL "nightly")
+    set(ENTERPRISE_BUILD_REPOSITORY "refs/head/${ARANGODB_VERSION_MAJOR}.${ARANGODB_VERSION_MINOR} ${COMMIT_SHORT}")
+    set(HAVE_ENTERPRISE_BUILD_REPOSITORY "1")
+  else ()
+    set(ENTERPRISE_BUILD_REPOSITORY "GIT FAILED TO RETRIEVE THE VERSION - UNSUPPORTED BUILD MODE")
+    set(HAVE_ENTERPRISE_BUILD_REPOSITORY "1")
+  endif()
 else ()
   set(ENTERPRISE_BUILD_REPOSITORY "")
   set(HAVE_ENTERPRISE_BUILD_REPOSITORY "0")
```
Comment (mpoeter) [inline CMakeLists.txt:392]: This is essentially a duplication of what we do above for the main repo. Can we keep this DRY and extract it to a separate function?

**Review by mpoeter — APPROVED** [review]
I would really appreciate if we could eliminate the code duplication (our CMake config is already a mess, I would prefer to not mess it up even more), otherwise LGTM.

**Review by mpoeter — APPROVED** [review]

**Review by KVS85 — APPROVED** [review]

**Pull request merged**